internal/controller/children.go

```diff
@@ -101,14 +102,34 @@ func (c ChildResource[T]) Reconcile(ctx context.Context, clnt client.Client, cr
 			if !ok {
 				return fmt.Errorf("Could not cast when reconciling")
 			}
+			events := v1.EventList{}
+			selector, err := fields.ParseSelector(fmt.Sprintf("involvedObject.name=%s,involvedObject.namespace=%s,involvedObject.kind=StatefulSet", current.ObjectMeta.Name, current.ObjectMeta.Namespace))
+			if err == nil {
+				if err := clnt.List(ctx, &events, &client.ListOptions{FieldSelector: selector}); err == nil {
+					for _, event := range events.Items {
+						if event.Reason == "FailedCreate" && strings.Contains(event.Message, "exceeded quota") {
+							statusCallback = func(status *amaltheadevv1alpha1.AmaltheaSessionStatus) {
+								status.State = amaltheadevv1alpha1.Failed
```

Comment:
suggestion: please add a test for finding the quota exceeded failure in the `When the session is failing` context in `test/e2e/general_test.go`. You can see examples for other tests for failing sessions there and follow the same pattern for this.